src/Trace/Attributes.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\ArrayTrait;
+
+/**
+ * This plain PHP class represents a Link resource. A set of attributes, each in
+ * the format [KEY]:[VALUE].
+ *
+ * @see https://cloud.google.com/trace/docs/reference/v2/rest/v2/Attributes
+ */
+class Attributes implements \JsonSerializable, \ArrayAccess
+{
+    /**
+     * @var array
+     */
+    private $attributes = [];
+
+    /**
+     * Callback for \ArrayAccess []= setter.
+     *
+     * @param string $offset
+     * @param mixed $value
+     */
+    public function offsetSet($offset, $value)
```

Comment:
I've made a general practice of marking the methods to implement `ArrayAccess` as private (`@access private`) and simply documenting in the class docblock that it supports array access, along with an example of using it in such a fashion. To demonstrate:

```
This class implements PHP's `ArrayAccess` interface, and thus supports all array-style read and assignment operations.

Example:
` ``
$attributes['foo'] = 'bar';
echo $attributes['foo'];
` ``
```